src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,51 @@
+import Popup from "./PopUp";
+
+export default class PopUpWithConfirmation extends Popup {
+    constructor(popupSelector) {
+        super(popupSelector);
+        this._popForm = this._popupElement.querySelector(".modal__form");
+        this._deletePopup = document.querySelector("#modal-confirm-delete");
+        this._deletePopupCloseBtn = this._popupElement.querySelector("#modal-button-close-confirm-delete");
+        this._deletePopupSubmit = this._popupElement.querySelector("#modal-button-confirm-delete");
+        //this._deletePopupSubmit.textContent = this._popSubmitBtnText;
+        this._deletePopUpSubmitText = this._deletePopupSubmit.textContent
+       
+    }
+    //open(){
+      //  super.open();
+        //this._deletePopupSubmit.removeAttribute("disabled");
+        //close(){
+            //this._popForm.reset();
+            //super.close();
+           
+       // }
+        setEventListeners(){
+            super.setEventListeners();
+            this._popForm.addEventListener("submit", (evt) => {
+                evt.preventDefault();
+                this._handleFormSubmit();
+            })
+        }
+       
+        setLoading(isLoading, loadingText="Deleting...") {
+            if(isLoading) {
+                this._hasConfirmed = true;
+                //return this._deletePopupSubmit.textContent  =  loadingText 
+               // his._deletePopupsubmit.textContent = this._deletePopUpSubmitText
+               this._deletePopupSubmit.textContent = "YES";
```

Comment:
If `isLoading` then you need to use `loadingText` in line 35


Actually it should work like that because `this._deletePopUpSubmitText` is stored in line 11 (you need to check why it doesn't work)

```js
  setLoading(isLoading, loadingText = "Deleting...") {
    if (isLoading) {
      this._deletePopupSubmit.textContent = loadingText;
    } else {
      this._deletePopUpSubmit.textContent = this._deletePopUpSubmitText;
    }
  }

```

Reply:
I have tried fixing this issue in multiple ways for the past few days and yet with no avail that's why I just decided to go to the route where I set the text manually. I renamed the "isLoading" to "isDeleting" and all the other naming components though to go with theme of its function. will that be okay?

Reply:
> I renamed the "isLoading" to "isDeleting" and all the other naming components though to go with theme of its function. will that be okay?

No. Your code should look like this


```js

  setDeleting = (isDeleting, deletingText = "Deleting...") => {
    if (isDeleting) {
      this._deletePopupSubmit.textContent = deletingText;
    } else {
      this._deletePopupSubmit.textContent = this._deletePopUpSubmitText;
    }
  };

```


If `isDeleting` then show loading othewise you set the default text


-------------------


I made method `setDeleting` as an **arrow** one and it worked


```js
  setDeleting = (isDeleting, deletingText = "Deleting...") => {


```


Please, make it the same way and you'll see